### PR TITLE
Alfoa/repetition related issues

### LINKED ISSUE #19: [DEFECT] Multiple Instantiations of CAPEX in repetition
--------
Defect Description
--------

**Describe the defect**
When using the "Repetitions" block and placing the end of the repetitions prior to the end of the global project lifetime the code throws an additional "CAPEX" at the end of the repetitions. 

##### What did you expect to see happen?
I would expect the code to just end all values at the end of the repetitions. 

##### What did you see instead?
I saw an additional CAPEX produced. See Year 10 below. This is using a repetition line as follows.   
Life_time> 10 /Life_time>
Repetitions> 1 /Repetitions> 

With a global lifetime of 30. 

CashFlow INFO (proj c_fl): Computing PROJECT cash flow for CashFlow "CA" ...
CashFlow INFO (proj comp): Project Cashflow for Component "Batt" CashFlow "CA":
CashFlow INFO (proj comp): Year, Time-Adjusted Value
CashFlow INFO (proj comp):    0:  3.375000000e+02
CashFlow INFO (proj comp):    1:  0.000000000e+00
CashFlow INFO (proj comp):    2:  0.000000000e+00
CashFlow INFO (proj comp):    3:  0.000000000e+00
CashFlow INFO (proj comp):    4:  0.000000000e+00
CashFlow INFO (proj comp):    5:  0.000000000e+00
CashFlow INFO (proj comp):    6:  0.000000000e+00
CashFlow INFO (proj comp):    7:  0.000000000e+00
CashFlow INFO (proj comp):    8:  0.000000000e+00
CashFlow INFO (proj comp):    9:  0.000000000e+00
**CashFlow INFO (proj comp):   10:  3.375000000e+02**
CashFlow INFO (proj comp):   11:  0.000000000e+00
CashFlow INFO (proj comp):   12:  0.000000000e+00
CashFlow INFO (proj comp):   13:  0.000000000e+00
CashFlow INFO (proj comp):   14:  0.000000000e+00
CashFlow INFO (proj comp):   15:  0.000000000e+00
CashFlow INFO (proj comp):   16:  0.000000000e+00
CashFlow INFO (proj comp):   17:  0.000000000e+00
CashFlow INFO (proj comp):   18:  0.000000000e+00
CashFlow INFO (proj comp):   19:  0.000000000e+00
CashFlow INFO (proj comp):   20:  0.000000000e+00
CashFlow INFO (proj comp):   21:  0.000000000e+00
CashFlow INFO (proj comp):   22:  0.000000000e+00
CashFlow INFO (proj comp):   23:  0.000000000e+00
CashFlow INFO (proj comp):   24:  0.000000000e+00
CashFlow INFO (proj comp):   25:  0.000000000e+00
CashFlow INFO (proj comp):   26:  0.000000000e+00
CashFlow INFO (proj comp):   27:  0.000000000e+00
CashFlow INFO (proj comp):   28:  0.000000000e+00
CashFlow INFO (proj comp):   29:  0.000000000e+00
CashFlow INFO (proj comp):   30:  0.000000000e+00

##### Do you have a suggested fix for the development team?

**Describe how to Reproduce**
1.  Create a TEAL input with a global lifetime = 30
2.  Create  a repetitions of 1
3.  Life_time = 10
4.  Look in the Capex system after the run. (no extraneous values occurred with the recurring tag. 

**Screenshots and Input Files**
Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

**Platform (please complete the following information):**
 - OS: Windows 10 using GITBASH

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### LINKED ISSUE #20: [DEFECT] Additional Instantiation of O&M when using repetition
--------
Defect Description
--------

**Describe the defect**
For a system lifetime of 10years within a 30 year project there were 11 O&M calculations as opposed to just 10. 
##### What did you expect to see happen?
I expected to see only 10 years worth of O&M

##### What did you see instead?
I saw 11 years worth. As seen below. 

CashFlow INFO (proj c_fl): Computing PROJECT cash flow for CashFlow "RE" ...
CashFlow INFO (proj comp): Project Cashflow for Component "Batt" CashFlow "RE":
CashFlow INFO (proj comp): Year, Time-Adjusted Value
CashFlow INFO (proj comp):    0: -0.000000000e+00
CashFlow INFO (proj comp):    1: -4.500000000e+01
CashFlow INFO (proj comp):    2: -4.500000000e+01
CashFlow INFO (proj comp):    3: -4.500000000e+01
CashFlow INFO (proj comp):    4: -4.500000000e+01
CashFlow INFO (proj comp):    5: -4.500000000e+01
CashFlow INFO (proj comp):    6: -4.500000000e+01
CashFlow INFO (proj comp):    7: -4.500000000e+01
CashFlow INFO (proj comp):    8: -4.500000000e+01
CashFlow INFO (proj comp):    9: -4.500000000e+01
CashFlow INFO (proj comp):   10: -4.500000000e+01
CashFlow INFO (proj comp):   11: -4.500000000e+01

##### Do you have a suggested fix for the development team?
Reduce the loop by 1. 

**Describe how to Reproduce**
1.  Run the input files in the attached zip.
[Error.zip](https://github.com/idaholab/TEAL/files/5631875/Error.zip)

Answer will be 4.602E+03
Answer **should** be 3.4822E+03
The **should** answer can be achieved if all three of the fixer variables are given an alpha of -1.0 to cancel out all the extra values in the system.

**Screenshots and Input Files**
Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

**Platform (please complete the following information):**
 - Version: [e.g. 22]
 - Dependencies Installation: [CONDA or PIP]

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address?
Closes #19 
Closes #20 

##### What are the significant changes in functionality due to this change request?

Addressed issue on repetition 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [x] 4. Automated Tests should pass.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [x] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by alfoa** [comment]
@PaulTalbot-INL @wangcj05  this modification will causes diffs in the plugins are based on TEAL. Please review it and see if it makes sense

**Comment by PaulTalbot-INL** [comment]
Do we have any way of knowing whether the old way was causing wrong results, and are changed in this new way?

**Comment by alfoa** [comment]
I think the issue  #20 is a good starting point.

**Comment by alfoa** [comment]
@PaulTalbot-INL  @wangcj05 decision?

File: src/main.py
Comment (wangcj05) [inline src/main.py:355]: I do not get the point here. From the user manual, it seems to me both start and end are included in the cashflow calculations.  @PaulTalbot-INL Could you provide some comment here?

Comment (alfoa) [reply src/main.py:355]: the first year is considered the "Construction year" that is added at the initialization of the analysis (look for ```self._projectTime```)

Comment (PaulTalbot-INL) [reply src/main.py:355]: Does this new change capture the last year of the project time?

This is my major concern here, I see no changes to existing tests. This tells me that either our testing coverage didn't cover replicated years, or this change has no impact on existing tests, which is also concerning.

Comment (alfoa) [reply src/main.py:355]: Yes it does. And yes, the tests were not covering this particular situation. This is the reason I added another test.

Comment (PaulTalbot-INL) [reply src/main.py:355]: Okay, if the last year is for sure covered, then I think this is a great PR and has my green light.

Comment (PaulTalbot-INL) [reply src/main.py:355]: It's surely beyond the scope of this PR, but I think we need a way to dump some of the stuff we print to screen (individual cash flows, etc) into a file so it can be reviewed; that way we could test and be sure all the correct years are being covered, the cash flows are combining correctly, and so on.

**Review by PaulTalbot-INL — APPROVED** [review]

File: src/main.py
Comment (alfoa) [reply src/main.py:355]: agree. I like this...

**Comment by alfoa** [comment]
@PaulTalbot-INL @wangcj05  can any of you merge this if we approve this?

**Review by wangcj05 — APPROVED** [review]

**Comment by wangcj05** [comment]
This is good to me. @alfoa Could you send out an email regarding this fix?

**Pull request merged**
